File: TestApplication/app/src/main/java/com/example/testapplication/ExceptionHandling.java
package com.example.testapplication;

class InvalidAgeException  extends Exception
{
    public InvalidAgeException (String str) {
        super(str);
    }
}

public class ExceptionHandling {

    static void validate (int age) throws InvalidAgeException{
        if(age < 18){
            throw new InvalidAgeException("Age is not valid to vote");
        }
        else {
            System.out.println("You can vote");
        }
    }

    public static void main(String[] args) {
        try {
            validate(13);
        } catch (InvalidAgeException ex) {
            System.out.println("Exception occurred: " + ex);
        }
        System.out.println("rest of the code...");
    }
}

File: KotlinPractice/app/src/main/java/com/example/kotlinpractice/exceptionHandling.kt
package com.example.kotlinpractice

class ExceptionHandling(msg: String) : Exception(msg)

fun checkAge(age: Int) {
    if (age < 18) throw ExceptionHandling("Invalid age")
    println("Valid age")
}

fun main() {
    try {
        checkAge(10)
    } catch (e: ExceptionHandling) {
        println(e.message)
    } finally {
        println("This is inside finally block")
    }
}

